src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -1405,8 +1404,8 @@ static inline int freeze_internal(JavaThread* current, intptr_t* const sp) {
     JvmtiSampledObjectAllocEventCollector jsoaec(false);
     fr.set_jvmti_event_collector(&jsoaec);
 
-    freeze_result res = fast ? fr.template try_freeze_fast<false>(sp)
-                             : fr.freeze_slow();
+    freeze_result res = can_freeze_fast(current) ? fr.template try_freeze_fast<false>(sp)
```

Comment:
we need a comment why not so some ambitious person doesn't think they can optimize it back ...